modules/genomic_browser/php/cnvbrowser.class.inc

```diff
@@ -0,0 +1,159 @@
+<?php declare(strict_types=1);
+
+namespace LORIS\genomic_browser;
+
+use LORIS\Data\Filters\HasAnyPermissionOrUserSiteMatch;
+use LORIS\genomic_browser\Provisioners\CnvProvisioner;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Middleware\ETagCalculator;
+
+/**
+ * The CnvBrowser Class.
+ *
+ * This class provides the php code to create the menu filter form that
+ * allows cnvbrowser to be processed as part of the genomic browser.
+ * Used for populating the Genomic Browser (CNV FilterDataTable).
+ *
+ * @category Loris
+ * @package  Genomic_Browser
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class CnvBrowser extends \NDB_Page implements ETagCalculator
+{
+
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+        if (isset($values['error'])) {
+            switch ($values['error']) {
+            case 'BadRequest':
+                return new \LORIS\Http\Response\JSON\BadRequest(
+                    'Failed to retrieve data for CNV.'
+                );
+            case 'Forbidden':
+                return new \LORIS\Http\Response\JSON\Forbidden();
+            default:
+                break;
+            }
+        }
```

Comment:
I don't understand those blocks. The frontend would send an error string to receive the corresponding HTTP status code?

Reply:
@xlecours oops I'll fix that.